modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpDownloaderTaskExecutor.java

```diff
@@ -125,6 +125,8 @@ protected GeoIpDownloader createTask(
             parentTaskId,
             headers
         );
+        clusterService.addListener(downloader);
```

Comment:
I think we should update the GeoIpDownloader to remove itself from the cluster listeners when it is cancelled.